docs/integration/KafkaIntegration.md

```diff
@@ -8,104 +8,164 @@ sidebar_position: 1
-- Which topics are available in Kafka sources and sinks
-- What schema versions are available for topic
-- How code completions and validations on types described by schemas should work
-- How messages will be serialized and deserialized
+Nussknacker uses schemas for code completions and validations of messages. 
+Schema of message can be described in [Avro Schema format](https://avro.apache.org/docs/#schemas) or [JSON Schema format](https://json-schema.org) (Confluent Schema Registry only)
 
-Currently, Nussknacker supports two implementations of schema registries: based on *Confluent Schema Registry* and based on *Azure Schema Registry*. 
-During runtime, it serializes and deserializes messages in the way that is compatible with standard Kafka serializers and deserializers 
-delivered by those schema registries providers. Thanks to that you should be able to send messages to Nussknacker and read messages 
-produced by Nussknacker using standard tooling available around those Schema Registries.
+Schemas are retrieved from Schema Registry - *Confluent Schema Registry* and *Azure Schema Registry* are supported.
 
-Given implementation is picked based on `schema.registry.url` property. By default, Confluent-based implementation is used. 
-For urls ended with `.servicebus.windows.net` Azure-based implementation is used.
+To preview schemas or add a new version, you can use tools available on your cloud platform or tools like [AKHQ](https://akhq.io)
+
+#### Association between schema with topic
+
+To properly present information about topics and version and to recognize which schema is assigned to version, Nussknacker follow conventions:
+- For Confluent-based implementation it uses [TopicNameStrategy for subject names](https://docs.confluent.io/platform/current/schema-registry/serdes-develop/index.html#subject-name-strategy).
+  It means that it looks for schemas available at `<topic-name>-(key or value)` *subject*. For example for topic `transactions`, it looks for schemas at `transactions-key` *subject* for key and `transactions-value` *subject* for value
+- In the Azure Schema Registry, *subject* concept doesn't exist - schemas are grouped by the same *schema name*. Because of that, Nussknacker introduces
+  own convention for association between schema and topic: *schema name* should be in format: `CamelCasedTopicNameKey` for keys and `CamelCasedTopicNameValue` for values.
+  For example for `input-events` topic, schema name should be named `InputEventsKey` for key or `InputEventsValue` for value. Be aware that it may require change of schema name
+  not only in Azure portal but also inside schema content - those names should be the same to make serialization works correctly
+
+## Connection and Authentication Configuration
 
-### Connection configuration
+Under the hood Nussknacker uses `kafkaProperties` to configure standard kafka client. It means that all standard Kafka client properties will be respected.
```

Comment:
I would omit the second sentence, what follows is a clear indication of some _not to be ignored_ configuration. 
Alternatively, I would replace word _respected_ with _provided_.

Reply:
I want to write here, that we respect many options that are not documented in our code, but that are documented in other documentations. Is it clear without this sentence?